wlauto/workloads/googleslides/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,469 @@
+            clickView(BY_TEXT, "Allow");
+        }
+
+        // Scroll through document list if necessary
+        UiScrollable list = new UiScrollable(new UiSelector().className("android.widget.ListView"));
+        list.scrollIntoView(new UiSelector().textContains(docName));
+        timer = new Timer();
+        timer.start();
+        clickView(BY_TEXT, docName);
+        clickView(BY_TEXT, "Open", CLASS_BUTTON, true);
+        timer.end();
+        results.put("open_document", timer);
+        sleep(5);
+
+        int centerY = getUiDevice().getDisplayHeight() / 2;
+        int centerX = getUiDevice().getDisplayWidth() / 2;
+        int slideIndex = 0;
+        String testTag;
+        Timer slideTimer;
+
+        // scroll forward in edit mode
+        startDumpsys(ACTIVITY_SLIDES);
+        timer = new Timer();
+        timer.start();
+        while (++slideIndex < slideCount) {
+            testTag = "slides_next_" + slideIndex;
+            startDumpsys(ACTIVITY_SLIDES);
+            slideTimer = new Timer();
+            slideTimer.start();
+            uiDeviceSwipeHorizontal(centerX + centerX/2, centerX - centerX/2, centerY);
+            slideTimer.end();
+            results.put(testTag, slideTimer);
+            endDumpsys(ACTIVITY_SLIDES, testTag);
+            sleep(1);
+        }
+        timer.end();
+        results.put("slides_forward", timer);
+        endDumpsys(ACTIVITY_SLIDES, "slides_forward");
+        sleep(1);
+
+        // scroll backward in edit mode
+        startDumpsys(ACTIVITY_SLIDES);
+        timer = new Timer();
+        timer.start();
+        while (--slideIndex > 0) {
+            testTag = "slides_previous_" + slideIndex;
+            startDumpsys(ACTIVITY_SLIDES);
+            slideTimer = new Timer();
+            slideTimer.start();
+            uiDeviceSwipeHorizontal(centerX - centerX/2, centerX + centerX/2, centerY);
+            slideTimer.end();
+            results.put(testTag, slideTimer);
+            endDumpsys(ACTIVITY_SLIDES, testTag);
+            sleep(1);
+        }
+        timer.end();
+        results.put("slides_reverse", timer);
+        endDumpsys(ACTIVITY_SLIDES, "slides_reverse");
+        sleep(1);
+
+        // scroll forward in slideshow mode
+        timer = new Timer();
+        timer.start();
+        clickView(BY_DESC, "Start slideshow", true);
+        timer.end();
+        results.put("open_slideshow", timer);
+
+        startDumpsys(ACTIVITY_SLIDES);
+        timer = new Timer();
+        timer.start();
+        while (++slideIndex < slideCount) {
+            testTag = "slideshow_next_" + slideIndex;
+            startDumpsys(ACTIVITY_SLIDES);
+            slideTimer = new Timer();
+            slideTimer.start();
+            uiDeviceSwipeHorizontal(centerX + centerX/2, centerX - centerX/2, centerY);
+            slideTimer.end();
+            results.put(testTag, slideTimer);
+            endDumpsys(ACTIVITY_SLIDES, testTag);
+            sleep(1);
```

Comment:
This is so that we get a perceptible wait-time before the next slide (i.e. real user behaviour). Otherwise it just swipes through the slides too quickly.